pandas/core/groupby/groupby.py

```diff
@@ -723,13 +724,24 @@ def _get_index(self, name):
     @cache_readonly
     def _selected_obj(self):
         # Note: _selected_obj is always just `self.obj` for SeriesGroupBy
-
-        if self._selection is None or isinstance(self.obj, Series):
-            if self._group_selection is not None:
-                return self.obj[self._group_selection]
+        if isinstance(self.obj, Series):
             return self.obj
-        else:
-            return self.obj[self._selection]
+
+        if self._selection is not None:
+            if is_hashable(self._selection):
+                # i.e. a single key, so selecting it will return a Series.
+                #  In this case, _obj_with_exclusions would wrap the key
+                #  in a list and return a single-column DataFrame.
+                return self.obj[self._selection]
```

Comment:
I think any time `self.obj[self._selection]` results in a Series will be when `self` is a SeriesGroupBy and `self.obj` is a Series, which is handled above.

If the frame has duplicate columns, then we can windup in the case where `self._selection` is hashable and `self` is a DataFrameGroupBy, but in this case I think `self.obj[self_selection]` will then agree with `self._obj_with_exclusions` (both frames).

So I think we can still return obj_with_exclusions here.

Reply:
That logic works for SeriesGroupBy/DataFrameGroupBy, but this code path is shared by Resampler, which defines _gotitem kludgily.  Until we change that, test_resample_groupby_agg_listlike would fail if we return obj_with_exclusions here

Reply:
> Until we change that

(this is part of why i pinged you about the non-unique columns in agg_dict_like.  If we can make sure to always pass `subset` to `_gotitem`, i think that makes it a lot easier to de-kludge Resample._gotitem)